oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/dsl/SampleFamily.java

```diff
@@ -310,6 +311,24 @@ public SampleFamily tag(Closure<?> cl) {
         );
     }
 
+    /* k8s add Tags*/
+    public SampleFamily retagByK8sMeta(String newLabelName, K8sRetagType type, String existingLabelName) {
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(newLabelName));
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(existingLabelName));
+        ExpressionParsingContext.get().ifPresent(ctx -> ctx.isRetagByK8sMeta = true);
+        if (this == EMPTY) {
+            return EMPTY;
+        }
+        switch (type) {
+            case Pod2Service:
+                return SampleFamily.build(this.context, K8sRetagOps.pod2Service(samples, newLabelName, existingLabelName));
+
+            default:
+                throw new UnsupportedOperationException(
+                    "Unsupported type: " + type + " in retagByK8sMeta() " + this.toString());
+        }
+    }
```

Comment:
Please consider K8sRetagType as an interface implementation. Then you could have `type.cover(param1, param2)` directly, rather than this switch case and throw an exception in the runtime.